### PR TITLE
Made sulu compatible with mysql 5.7

### PR DESCRIPTION
**Pull request by marcelmoosbrugger**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes https://github.com/sulu/sulu/issues/1797, fixes https://github.com/sulu/sulu/issues/2645
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

see title.

### CODE REVIEW

**Comment by marcelmoosbrugger** [comment]
@danrot This PR is ready for review.

**Comment by marcelmoosbrugger** [comment]
Note that travis does not execute the tests on Mysql 5.7. It's not yet supported (they have an open ticket for that). I created an issue so we don't forget to update travis:
https://github.com/sulu/sulu/issues/2702

File: src/Sulu/Bundle/MediaBundle/Entity/CollectionRepository.php [hunk 1]
```diff
@@ -297,4 +271,67 @@ public function findCollectionTypeById($id)
...
+     *
+     * @return Query
+     */
+    private function getIdsQuery(
+        $depth = 0,
+        $filter = [],
+        CollectionInterface $collection = null,
+        $select = 'collection.id'
+    ) {
+        $queryBuilder = $this->createQueryBuilder('collection')->select($select)
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Entity/CollectionRepository.php]: `->select` call should also go into an own line.

File: src/Sulu/Bundle/MediaBundle/Entity/CollectionRepository.php [hunk 2]
```diff
@@ -71,79 +71,53 @@ public function findCollectionSet(
...
-        } catch (NoResultException $ex) {
-            return [];
+    /**
+     * {@inheritdoc}
+     */
+    public function count($depth = 0, $filter = [], CollectionInterface $collection = null)
+    {
+        $query = $this->getIdsQuery($depth, $filter, $collection, 'COUNT(DISTINCT collection.id)');
+        try {
+            $result = $query->getSingleResult()[1];
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Entity/CollectionRepository.php]: Why `[1]`?

File: src/Sulu/Bundle/MediaBundle/Entity/CollectionRepository.php [hunk 3]
```diff
@@ -71,79 +71,53 @@ public function findCollectionSet(
...
+            ->addSelect('parentMeta')
+            ->addSelect('collectionChildren')
+            ->leftJoin('collection.meta', 'collectionMeta')
+            ->leftJoin('collection.defaultMeta', 'defaultMeta')
+            ->leftJoin('collection.type', 'collectionType')
+            ->leftJoin('collection.parent', 'collectionParent')
+            ->leftJoin('collection.children', 'collectionChildren')
+            ->leftJoin('collectionParent.meta', 'parentMeta')
+            ->where('collection.id IN (:ids)');
+        $queryBuilder->setParameter('ids', $ids);
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Entity/CollectionRepository.php]: Writing the `$queryBuilder` variable in this line is a bit inconsistent.

File: src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php [hunk 4]
```diff
@@ -304,24 +309,18 @@ protected function getCategoryListRepresentation(Request $request, $parentKey =
...
-        $listBuilder->addGroupBy($fieldDescriptors['id']);
+        $listBuilder->addSelectField($fieldDescriptors['lft']);
+        $listBuilder->addSelectField($fieldDescriptors['rgt']);
 
         if ($parentKey !== null) {
             $this->addParentSelector($parentKey, $listBuilder);
         }
 
-        // FIXME: don't do this.
-        $listBuilder->limit(100000);
```
Comment (danrot) [inline src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php]: I've got the feeling that you can't delete that line... Now only 10 elements are returned, and the children are not loaded. So if you have 10 root categories none of the children are loaded. Maybe it works when you set the limit to `null`? The no limit at all should be applied, and it is not a magic number like now. The categories should be improved in that way quite soon anyway.

**Pull request merged**